README.md

```diff
@@ -1 +1,118 @@
-# replace this
\ No newline at end of file
+#### CDK Construct Analyzer
+
+`@cdklabs/cdk-construct-analyzer` is a CLI and library for evaluating the quality of construct libraries.
+
+It calculates a single score (0–100) based on three equally weighted aspects:
+
+* **Maintenance**: Is the project actively maintained and are owners/maintainers responsive?
+* **Quality**: Does the project have good docs, tests, linting, and hygiene?
+* **Popularity**: How widely is the library adopted in the community?
+
+Each package is scored on their latest version. Scores are unlikely to change drasically between versions.
+
+#### CLI Usage
+
+```> cdk-construct-analyzer --help
+
+Usage: cdk-construct-analyzer [package] [options]
+
+Arguments:
+  package   Name of the construct package to score (Scored on the latest version)
+
+Options:
+ --verbose  Show detailed breakdown of signals
+ --help     Show this help message
+```
+
+You can run it locally on any library published to npm by providing its package name:
+
+```> cdk-construct-analyzer cdk-ecr-codedeploy
+
+LIBRARY: @cdklabs/cdk-ecr-codedeploy
+VERSION: 0.0.421
+
+OVERALL SCORE: 76/100
+
+---
+
+SUBSCORES
+  Maintenance :            66/100
+  Quality     :            75/100
+  Popularity  :            88/100
+```
+
+Add `--verbose` for a detailed breakdown:
+
+```> cdk-construct-analyzer cdk-ecr-codedeploy --verbose
+
+LIBRARY: cdk-ecr-codedeploy
+VERSION: 0.0.421
+
+OVERALL SCORE: 76/100
+
+---
+
+SUBSCORES
+  Maintenance :            66/100
+  Quality     :            75/100
+  Popularity  :            88/100
+  
+---
+
+=== Maintenance ===                               SCORE  WEIGHT
+— Time to first response......................... ★★☆☆☆    3
+— Provenance Verification ....................... ★★★★★    3
+— Number of Contributors ........................ ★★★★☆    2
+
+=== Quality ===                                   SCORE  WEIGHT
+— Documentation Completeness .................... ★★★★★    3
+
+=== Popularity ===                                SCORE  WEIGHT
+— Weekly Downloads .............................. ★★★★★    3
+— Repo stars .................................... ★★★★☆    2
+— Contributors .................................. ★★★★☆    1
+```
```

Comment:
another section here about programmatic access -- i.e. i am an author and i wish to embed your Typescript functions that you've exposed in my application. how would i use it?